src/components/Book/BookList.tsx

```diff
@@ -42,19 +43,38 @@ const BookList = ({ books, penname, isOwner, isArchived }: props) => {
 
   return (
     <div className="grid grid-cols-4 gap-x-8 gap-y-6">
-      {books.map((book) => (
-        <Book
-          key={book.id}
-          id={book.id}
-          title={book.title}
-          coverImage={book.coverImage}
-          description={book.description}
-          isOwner={book.isOwner}
-          status={book.status}
-          like={book.chapters.reduce((acc, curr) => acc + curr._count.likes, 0)}
-          read={book.chapters.reduce((acc, curr) => acc + curr._count.views, 0)}
-        />
-      ))}
+      {books.map(
+        (book) =>
+          ((isOwner && !book.isRejected) ||
+            (!isOwner &&
+              (book.status === BookStatus.PUBLISHED ||
+                book.status === BookStatus.COMPLETED))) && (
+            <Book
+              key={book.id}
+              id={book.id}
+              title={book.title}
+              ownerPenname={
+                book.owners.find(
+                  (data) => data.status === BookOwnerStatus.OWNER
+                )?.user.penname || null
+              }
+              coverImage={book.coverImage}
+              description={book.description}
+              isOwner={book.isOwner}
+              isCollaborator={book.isCollborator}
+              isInvitee={book.isInvited}
+              status={book.status}
+              like={book.chapters.reduce(
+                (acc, curr) => acc + curr._count.likes,
+                0
+              )}
+              read={book.chapters.reduce(
+                (acc, curr) => acc + curr._count.views,
+                0
+              )}
+            />
+          )
+      )}
```

Comment:
extract this to a new function, use book.map like this will have some undefined in the array --> use filter before map